code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -111,3 +131,12 @@
 /obj/item/reagent_container/hypospray/oxycodone/New()
 	. = ..()
 	update_icon()
+
+/obj/item/reagent_container/hypospray/attack_self(mob/user) //set amount_per_transfer_from_this
+	var/str = copytext(reject_bad_text(input(user,"Label text?", "Set label", "")), 1, MAX_NAME_LEN)
+	if(!str || !length(str))
+		to_chat(user, "<span class='notice'>Invalid text.</span>")
+		return
+	user.visible_message("<span class='notice'>[user] labels [src] as \"[str]\".</span>", \
+						 "<span class='notice'>You label [src] as \"[str]\".</span>")
```

Comment:
Normally user.visible_message() has a standard view range of 7. You should modify it as such action shouldn't be so noticeable.